rust-runtime/aws-smithy-runtime-api/src/http/response.rs

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Http Response Types
+
+use crate::http::{HeaderValue, Headers, HttpError};
+use aws_smithy_types::body::SdkBody;
+use http as http0;
+use http0::{Extensions, HeaderMap};
+use std::fmt;
+
+/// HTTP response status code
+#[derive(Copy, Clone, Debug)]
+pub struct StatusCode(u16);
+
+impl StatusCode {
+    /// True if this is a successful response code (200, 201, etc)
+    pub fn is_success(self) -> bool {
+        (200..300).contains(&self.0)
+    }
+
+    /// True if this response code is a client error (4xx)
+    pub fn is_client_error(self) -> bool {
+        (400..500).contains(&self.0)
+    }
+
+    /// True if this response code is a server error (5xx)
+    pub fn is_server_error(self) -> bool {
+        (500..600).contains(&self.0)
+    }
+
+    /// Return the value of this status code as a `u16`.
+    pub fn as_u16(self) -> u16 {
+        self.0
+    }
+}
+
+impl TryFrom<u16> for StatusCode {
+    type Error = HttpError;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        if (100..1000).contains(&value) {
+            Ok(StatusCode(value))
+        } else {
+            Err(HttpError::invalid_status_code())
+        }
+    }
+}
+
+impl From<StatusCode> for u16 {
+    fn from(value: StatusCode) -> Self {
+        value.0
+    }
+}
+
+impl fmt::Display for StatusCode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+/// An HTTP Response Type
+#[derive(Debug)]
+pub struct Response<B = SdkBody> {
+    status: StatusCode,
+    headers: Headers,
+    body: B,
+    extensions: Extensions,
+}
+
+impl<B> Response<B> {
```

Comment:
I'm uncertain about our handling of streaming responses; however, I presume this change won't impact that, correct?

Reply:
This doesn't make any changes to streaming.